libwallet/assets/btc/txandblocknotifications.go

```diff
@@ -96,9 +96,9 @@ func (asset *Asset) AddTxAndBlockNotificationListener(txAndBlockNotificationList
 	}
 
 	asset.txAndBlockNotificationListeners[uniqueIdentifier] = &sharedW.TxAndBlockNotificationListener{
-		OnTransaction: func(transaction *sharedW.Transaction) {
+		OnTransaction: func(walletID int, transaction *sharedW.Transaction) {
 			if txAndBlockNotificationListener.OnTransaction != nil {
-				go txAndBlockNotificationListener.OnTransaction(transaction)
+				go txAndBlockNotificationListener.OnTransaction(walletID, transaction)
```

Comment:
```suggestion
				go txAndBlockNotificationListener.OnTransaction(asset.ID, transaction)
```

can't we get the id fromt he calling asset?

Reply:
But I think it's better we pass it in when executing the actual function.